docs/day-2-operations/Backups.md

```diff
@@ -0,0 +1,272 @@
+* Config: This is configuration information about Services, Rate-limits, etc that is synchronized from `System` via an internal API. This is NOT the source of truth of this info, `System` and `system-mysql` is.
+* AuthKeys: Storage of OAuth keys created directly in `Backend`. This is the source of truth for this information.
+* Queues: Queues of Background Jobs to be executed by worker processses. These are ephemeral and are deleted once processed.
+
+### `system-redis`
+This contains Queues for jobs to be processed in background. These are ephemeral and are deleted once processed.
+
+## Backup Procedures
+
+### `system-mysql`
+
+1. Execute MySQL Backup Command
+   ```
+   oc rsh $(oc get pods -l 'deploymentConfig=system-mysql' -o json | jq -r '.items[0].metadata.name') bash -c 'export MYSQL_PWD=${MYSQL_ROOT_PASSWORD}; mysqldump --single-transaction -hsystem-mysql -uroot system' | gzip > system-mysql-backup.gz
+   ```
+
+### `system-storage`
+
+1. Archive the system-storage files to another storage.
+    ```
+    oc rsync $(oc get pods -l 'deploymentConfig=system-app' -o json | jq '.items[0].metadata.name' -r):/opt/system/public/system ./local/dir
+    ```
+
+### `zync-database`
+
+1. Execute Postgres Backup Command
+   ```
+    oc rsh $(oc get pods -l 'deploymentConfig=zync-database' -o json | jq '.items[0].metadata.name' -r) bash -c 'pg_dumpall -c --if-exists' | gzip > zync-database-backup.gz
+    ```
+
+### `backend-redis`
+
+1. Backup the dump.rb file from redis
+   ```
+   oc cp $(oc get pods -l 'deploymentConfig=backend-redis' -o json | jq '.items[0].metadata.name' -r):/var/lib/redis/data/dump.rdb ./backend-redis-dump.rdb
+   ```
+
+### `system-redis`
+
+1. Backup the dump.rb file from redis
+
+   ```
+   oc cp $(oc get pods -l 'deploymentConfig=system-redis' -o json | jq '.items[0].metadata.name' -r):/var/lib/redis/data/dump.rdb ./system-redis-dump.rdb
+   ```
+
+## Restore Procedures
+
+### `system-mysql`
+
+1. Copy the MySQL dump to the system-mysql pod
+   ```
+   oc cp ./system-mysql-backup.gz $(oc get pods -l 'deploymentConfig=system-mysql' -o json | jq '.items[0].metadata.name' -r):/var/lib/mysql
+   ```
+2. Decompress the Backup File
+    ```
+    oc rsh $(oc get pods -l 'deploymentConfig=system-mysql' -o json | jq -r '.items[0].metadata.name') bash -c 'gzip -d ${HOME}/system-mysql-backup.gz'
+    ```
+3. Restore the MySQL DB Backup file
+   ```
+   oc rsh $(oc get pods -l 'deploymentConfig=system-mysql' -o json | jq -r '.items[0].metadata.name') bash -c 'export MYSQL_PWD=${MYSQL_ROOT_PASSWORD}; mysql -R -hsystem-mysql -uroot system < ${HOME}/system-mysql-backup'
+   ```
+
+### `system-storage`
+
+1. Restore the Backup file to system-storage
+   ```
+   oc rsync ./local/dir $(oc get pods -l 'deploymentConfig=system-app' -o json | jq '.items[0].metadata.name' -r):/opt/system/public/system
```

Comment:
Looks like directory here would be wrong too.

Reply:
s3 deployment?

Reply:
aaa true